server/src/main/java/org/opensearch/index/translog/transfer/FileSnapshot.java

```diff
@@ -38,12 +39,15 @@ public class FileSnapshot implements Closeable {
     private Path path;
     @Nullable
     private byte[] content;
+    @Nullable
+    private Map<String, String> metadata;
 
-    private FileSnapshot(Path path) throws IOException {
+    private FileSnapshot(Path path, Map<String, String> metadata) throws IOException {
         Objects.requireNonNull(path);
         this.name = path.getFileName().toString();
         this.path = path;
         this.fileChannel = FileChannel.open(path, StandardOpenOption.READ);
+        this.metadata = metadata;
     }
 
     private FileSnapshot(String name, byte[] content) {
```

Comment:
can we call `this(args..)` here?

Reply:
we can not combined these two constructor into one as both of the constructor's have different args..

Reply:
Why not overload?